Content.Client/Recycling/RecyclerVisualizerSystem.cs

```diff
@@ -0,0 +1,54 @@
+using Content.Shared.Conveyor;
+using Content.Shared.Recycling;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Recycling;
+
+public sealed class RecyclerVisualizerSystem : VisualizerSystem<RecyclerVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RecyclerVisualizerComponent, ComponentInit>(OnInit);
+    }
+
+    private void OnInit(EntityUid uid, RecyclerVisualizerComponent comp, ComponentInit args)
+    {
+        if (!TryComp(uid, out SpriteComponent? sprite) ||
+            !TryComp(uid, out AppearanceComponent? appearance))
+        {
+            return;
+        }
+
+        UpdateAppearance(uid, comp, appearance, sprite);
```

Comment:
make resolves inside UpdateAppearance for sprite and appearance instead of getting them here